### ISSUE TITLE
Request for SCS Service Registry ReactiveDiscoveryClient

### ISSUE THREAD

**Issue by maximusfloydus** [pivotal-cf/spring-cloud-services-starters#141]
Are there plans to create a OAuth2 Reactive Eureka Discovery client that can be used with SCS Service Registry?  If not, any pointers on what/where I should override things to use [EurekaReactiveDiscoveryClient](https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/reactive/EurekaReactiveDiscoveryClient.java}.  I am currently blocked from using a Jersery1 client on my classpath.

**Comment by dyroberts** [comment]
Hi @maximusfloydus. I think you've kind of asked two questions here. :) 

1. Reactive Discovery Client
I've looked into this a bit as I was not familiar with the `EurekaReactiveDiscoveryClient`. It's quite new in spring-cloud-netflix. As it turns out, it's really just a wrapper around the existing, blocking, `EurekaClient` and underlying `EurekaHttpClient` interfaces. This will probably be true until the underlying netflix library includes a reactive api.

2. Dropping Jersey
As you've noticed, the SCS client libraries rely on an extension point in the jersey-based `JerseyApplicationClient` for OAuth2 support. It would be nice to migrate away from jersey and use something like what you're suggesting. I believe the `RestTemplateEurekaHttpClient` and its associated factories, auto-config etc. would be a good example of how that would be done. If you were going to have a crack at that, I'd suggest using `WebClient` instead of `RestTemplate`, especially if you're working with webflux apps. We'd happily consider a PR like that to this repo. Alternatively, I'd be interested to try it myself when it fits with out team's priorities.

**Comment by dyroberts** [comment]
cc @csterwa

**Comment by mbwhelan** [comment]
Yes, I think a new Oauth2 Eureka Client based on WebClient would be best.  @dyroberts I am also going to see if I get this on my teams schedule, unless you have beaten me to it.

**Comment by dyroberts** [comment]
@maximusfloydus @mbwhelan Please go ahead with the PR! We always welcome contributions.

**Comment by stwomack** [comment]
@dyroberts Customer has decided not to work on this themselves. Any chance your team could make the bandwith to work on it? /cc @xyloman

**Comment by dyroberts** [comment]
@stwomack I've moved it to our backlog.

**Comment by stwomack** [comment]
@dyroberts Is there a jira issue or something else I could check in on by chance?

**Comment by dyroberts** [comment]
@stwomack @maximusfloydus @mbwhelan @xyloman I've merged an update to master, 1347e7d4b6259e39a7c2548a7cb73cafc204e0f8, that drops the jersey-client dependency.

@maximusfloydus @mbwhelan It turns out spring-cloud-netflix has already implemented a WebClient based EurekaHttpClient; but it won't be in general availability until the next major release, which will be this October as far as I can tell. I've switched the spring-cloud-services-starters to use `RestTemplateEurekaHttpClient` and dropped the jersey dependency. Please let me know if this covers your needs for now. I will be able to get a release out with this change later this week.
